lib/src/fnc/util/http/mod.rs

```diff
@@ -1,20 +1,24 @@
 use crate::ctx::Context;
 use crate::err::Error;
+use crate::http::header::CONTENT_TYPE;
+use crate::http::{Client, Request, Response};
 use crate::sql::object::Object;
 use crate::sql::strand::Strand;
 use crate::sql::value::Value;
 use crate::sql::{json, Bytes};
-use reqwest::header::CONTENT_TYPE;
-use reqwest::{Client, RequestBuilder, Response};
 
 pub(crate) fn uri_is_valid(uri: &str) -> bool {
-	reqwest::Url::parse(uri).is_ok()
+	url::Url::parse(uri).is_ok()
 }
 
-fn encode_body(req: RequestBuilder, body: Value) -> RequestBuilder {
+fn encode_body(req: Request, body: Value) -> Request {
 	match body {
-		Value::Bytes(bytes) => req.header(CONTENT_TYPE, "application/octet-stream").body(bytes.0),
-		_ if body.is_some() => req.json(&body.into_json()),
+		Value::Bytes(bytes) => req
+			.header(CONTENT_TYPE, "application/octet-stream")
+			.expect("header should be correct")
+			.body(bytes.0),
+		// TODO: Check if this is actually correct.
```

Comment:
Is this covered by tests? if it is then we can remove the todo. If not then we can write a test.

Reply:
For the header part. it should be guaranteed to never panic. "application/octet-stream" is a correct header value and CONTENT_TYPE is a header name supplied by the html crate so it should always be correct.

The TODO is about the line below serializing into json, which I believe should never return an error but I still need to validate that.